conductr_cli/conduct_load.py

```diff
@@ -95,15 +95,42 @@ def get_v_1_0_payload(bundle_name, bundle_file, bundle_configuration):
     ]
 
 
-def get_v_1_1_payload(bundle_name, bundle_file, bundle_configuration):
-    return [
-        ('nrOfCpus', bundle_configuration(ConfigTree.get_string, 'nrOfCpus')),
-        ('memory', bundle_configuration(ConfigTree.get_string, 'memory')),
-        ('diskSpace', bundle_configuration(ConfigTree.get_string, 'diskSpace')),
-        ('roles', ' '.join(bundle_configuration(ConfigTree.get_list, 'roles'))),
-        ('bundleName', bundle_configuration(ConfigTree.get_string, 'name')),
-        ('system', bundle_configuration(ConfigTree.get_string, 'system')),
-        ('systemVersion', bundle_configuration(ConfigTree.get_string, 'systemVersion')),
-        ('compatibilityVersion', bundle_configuration(ConfigTree.get_string, 'compatibilityVersion')),
-        ('bundle', (bundle_name, open(bundle_file, 'rb')))
-    ]
+def load_v2(args):
+    print('Retrieving bundle...')
+    bundle_name, bundle_url = get_url(args.bundle)
+    bundle_file, bundle_headers = urlretrieve(bundle_url)
+    bundle_conf = bundle_utils.zip_entry('bundle.conf', bundle_file)
+
+    if bundle_conf is None:
+        raise MalformedBundleError('Unable to find bundle.conf within the bundle file')
+    else:
+        configuration_file, bundle_conf_overlay = (None, None)
+        if args.configuration is not None:
+            print('Retrieving configuration...')
+            configuration_name, configuration_url = get_url(args.configuration)
+            configuration_file, configuration_headers = urlretrieve(configuration_url)
+            bundle_conf_overlay = bundle_utils.zip_entry('bundle.conf', configuration_file)
+
+        files = [('bundleConf', ('bundle.conf', bundle_conf))]
+        if bundle_conf_overlay is not None:
+            files.append(('bundleConfOverlay', ('bundle.conf', bundle_conf_overlay)))
```

Comment:
Don't we want to add the conf overlay part still, even if there's no overlay conf (in which case it is just a part with no filename)?